File: android/src/main/kotlin/com/expresspay/fluttersdk/ExpresspaySDKEventChannels.kt
package com.expresspay.fluttersdk

import android.content.Context
import com.expresspay.fluttersdk.eventhandlers.CardPayEventHandler
import com.expresspay.fluttersdk.eventhandlers.SaleEventHandler
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.EventChannel

/** ExpresspaySdkPlugin */
class ExpresspaySDKEventChannels{

  var cardpay:EventChannel? = null;
  var applepay:EventChannel? = null;

  var sale:EventChannel? = null;
  var recurringSale:EventChannel? = null;
  var capture:EventChannel? = null;
  var creditVoid:EventChannel? = null;
  var transactionStatus:EventChannel? = null;
  var transactionDetail:EventChannel? = null;
  var transactionLogs:EventChannel? = null;


  public fun initiate(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding, context: Context) {
    val messenger = flutterPluginBinding.binaryMessenger

    cardpay = EventChannel(messenger,"com.expresspay.sdk.cardpay")
    sale = EventChannel(messenger,"com.expresspay.sdk.sale")
    recurringSale = EventChannel(messenger,"com.expresspay.sdk.recurringsale")
    capture = EventChannel(messenger,"com.expresspay.sdk.capture")
    creditVoid = EventChannel(messenger,"com.expresspay.sdk.creditvoid")
    transactionStatus = EventChannel(messenger,"com.expresspay.sdk.transactionstatus")
    transactionDetail = EventChannel(messenger,"com.expresspay.sdk.transactiondetail")
    transactionLogs = EventChannel(messenger,"com.expresspay.sdk.transactionlogs")

    cardpay?.setStreamHandler(CardPayEventHandler(context))
    sale?.setStreamHandler(SaleEventHandler())

  }
}

File: ios/Classes/ExpresspaySdkEventChannels.swift
//
//  ExpressPaySDKEventChannels.swift
//  expresspay_sdk
//
//  Created by Zohaib Kambrani on 02/03/2023.
//

import Foundation
import Flutter
import UIKit



public class ExpressPaySDKEventChannels: NSObject{
    
    var cardpay:FlutterEventChannel? = nil;
    var applepay:FlutterEventChannel? = nil;
    
    var sale:FlutterEventChannel? = nil;
    var recurringSale:FlutterEventChannel? = nil;
    var capture:FlutterEventChannel? = nil;
    var creditVoid:FlutterEventChannel? = nil;
    var transactionStatus:FlutterEventChannel? = nil;
    var transactionDetail:FlutterEventChannel? = nil;
    var transactionLogs:FlutterEventChannel? = nil;

    
    public func initiate(with flutterViewController: FlutterViewController) {
        
        let messenger = flutterViewController.binaryMessenger
        
        cardpay = FlutterEventChannel(name: "com.expresspay.sdk.cardpay", binaryMessenger: messenger)
        applepay = FlutterEventChannel(name: "com.expresspay.sdk.applepay", binaryMessenger: messenger)
        
        sale = FlutterEventChannel(name: "com.expresspay.sdk.sale", binaryMessenger: messenger)
        recurringSale = FlutterEventChannel(name: "com.expresspay.sdk.recurringsale", binaryMessenger: messenger)
        capture = FlutterEventChannel(name: "com.expresspay.sdk.capture", binaryMessenger: messenger)
        creditVoid = FlutterEventChannel(name: "com.expresspay.sdk.creditvoid", binaryMessenger: messenger)
        transactionStatus = FlutterEventChannel(name: "com.expresspay.sdk.transactionstatus", binaryMessenger: messenger)
        transactionDetail = FlutterEventChannel(name: "com.expresspay.sdk.transactiondetail", binaryMessenger: messenger)
        transactionLogs = FlutterEventChannel(name: "com.expresspay.sdk.transactionlogs", binaryMessenger: messenger)
        
        
        
        cardpay?.setStreamHandler(CardPayEventHandler())
        applepay?.setStreamHandler(ApplePayEventHandler())
        
        sale?.setStreamHandler(SaleEventHandler())
        recurringSale?.setStreamHandler(RecurringSaleEventHandler())
        capture?.setStreamHandler(CaptureEventHandler())
        creditVoid?.setStreamHandler(CreditVoidEventHandler())
        transactionDetail?.setStreamHandler(TransactionDetailEventHandler())
        transactionStatus?.setStreamHandler(TransactionStatusEventHandler())

    }
    
}

